lib/BeaconDevice.js

```diff
@@ -135,6 +148,68 @@ class BeaconDevice extends Homey.Device {
             this.changeDetected = 0;
         }
     }
+
+    /**
+     *  New method that determines if a beacon advertisement is related to a 
+     *  device defined in Homey. It also update capability values, setting
+     *  (info) values and store values.
+     */
+    foundInAdvertisements(driver, advertisements) {
```

Comment:
I think the `foundInAdvertisements` should be in the `Driver`. That class is responsible for finding the correct type. The metadata extraction and storing of the specific data should be part of the `Device`.

Reply:
I have refactored the code as you suggested.